### PR TITLE
Revert "Fixes carbons with no eyes having a flash overlay applied when they're flashed."

### PR DESCRIPTION
**Pull request by Barhandar**
Reverts tgstation/tgstation#43175

NOEYES is actually "no eye sprites, they still have actual eyes lol". The PR being reverted makes moths flashproof, which is opposite of what is intended; until it can be fixed properly (by giving moths eye sprites + renaming current NOEYES into NOEYESPRITES or something) this should be reverted.

### CODE REVIEW

**Comment by ExcessiveUseOfCobblestone** [comment]
uh I thought it was the other way around?

I was under the impression the tag was actually for species without eyes, moth is just misusing it to avoid sprites.

**Comment by Barhandar** [comment]
I don't actually know, going off of what Calev and others have said. Either way, it'll need a rename/split for proper fix, and right now whatever weird species (abductors, mushes, cultgolems) have genuine lack of eyes being flashable is better issue than (roundstart selectable) moths being completely flashproof.

Also, _nightmares_ have NOEYES, and given their aversion to light they absolutely should be flashable and extra hurt by flashes.

**Comment by Shapsy** [comment]
based

File: code/modules/mob/living/carbon/carbon_defense.dm [hunk 1]
```diff
@@ -287,16 +287,13 @@
...
-		return
-	var/obj/item/organ/eyes/eyes = getorganslot(ORGAN_SLOT_EYES)
-	if(!eyes) //can't flash what can't see!
-		return
-
 	. = ..()
 
 	var/damage = intensity - get_eye_protection()
 	if(.) // we've been flashed
+		var/obj/item/organ/eyes/eyes = getorganslot(ORGAN_SLOT_EYES)
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/mob/living/carbon/carbon_defense.dm]: why was this moved

Comment (Barhandar) [reply code/modules/mob/living/carbon/carbon_defense.dm]: Ask Calev, I literally just pressed "revert" button on the merge message in the old PR on github site.

Comment (ExcessiveUseOfCobblestone) [reply code/modules/mob/living/carbon/carbon_defense.dm]: prob should stay at top even if you're reverting DNA chance. No point in even running inheritance if you're just going to lolnope it imo.

Comment (Barhandar) [reply code/modules/mob/living/carbon/carbon_defense.dm]: Good point, albeit it could be removed entirely - it's never actually reached, get_eye_protection() has a check for eyes too and will fail the check in parent if they're missing.

Comment (ExcessiveUseOfCobblestone) [reply code/modules/mob/living/carbon/carbon_defense.dm]: even better

**Comment by optimumtact** [comment]
You know moths being unflashable is really not that big of a deal right now

**Comment by VileBeggar** [comment]
> 
> 
> You know moths being unflashable is really not that big of a deal right now
it still affects gameplay for rev rounds. most headrevs logically conclude that moths are flashable, so they try to flash a moth and they end up getting revealed and probably dunked by security.

**Comment by ATH1909** [comment]
> > You know moths being unflashable is really not that big of a deal right now
> 
> it still affects gameplay for rev rounds. most headrevs logically conclude (with their big brains) that moths are flashable, so they try to flash a moth and then they fail, end up getting revealed and probably dunked by security.

I can confirm this; I was able to call out two headrevs in the same round as a moth doctor just because they tried and failed to convert me. One of my call-outs happened within 30 seconds of the round starting, which must have felt really bad for all of the revheads (not just the one(s) that tried to flash me).

**Review by ShizCalev — CHANGES_REQUESTED** [review]
This is just bandaiding over the issue that the races are using the flag incorrectly, which results in them not regenerating eyes correctly when adminhealed iirc.

Refer to #43261 on the proper way to fix this.

**Comment by ExcessiveUseOfCobblestone** [comment]
Just remove the noeye flag and if mothpeople cry about their character visuals tell them to fix the issue lol.

**Comment by Barhandar** [comment]
It is bandaiding, the point of which is to _have the game behave as expected while the correct solution is being made by you_, because nobody else seems to be competent enough to do it and it keeps resulting in lolnoped revs and moths ignoring welding and brig's flashes (not that people are using them nowadays).

**Comment by ShizCalev** [comment]
you're literally reintroducing the bug where species that -don't- have eyes -can- be flashed though.

**Comment by ExcessiveUseOfCobblestone** [comment]
> Just remove the noeye flag (for mothpeople) and if mothpeople cry about their character visuals tell them to fix the issue lol.

**Comment by Barhandar** [comment]
>you're literally reintroducing the bug where species that -don't- have eyes -can- be flashed though, which is obviously the game behaving in a way not expected.

They're not roundstart species, and hence extremely rarely seen. I've never witnessed mushies, for example. For golems and nightmares flashproofing is a bug, and abductors _probably_ already have flash protection on their gear. Either way, #43661
